src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let stateClone = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateClone, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete stateClone[key];
```

Comment:
You mutate one initial object all the time
Much better is to create a copy and mutate it on each action

Reply:
I don't modify the initial state. When I create a new copy on each action. 
`Should handle multiple types 
Should handle a long list of types `
These two tests don't pass.